docs/serving/using-knative-services.md

```diff
@@ -0,0 +1,233 @@
+
+## Creating a Knative Service
+To create an application, you need to create a YAML file that defines a Knative Service.
+This YAML file specifies metadata about the application, points to the hosted image of the app for deployment, and allows the deployment to be configured.
+
+This guide uses the [Hello World sample app in Go](https://knative.dev/docs/serving/samples/hello-world/helloworld-go) to demonstrate the structure of a Service YAML file and the basic workflow for deploying an app, but these steps can be adapted for your own application if you have an image of it available on Docker Hub, Google Container Registry, or another container image registry.
+
+The Hello World sample app does the following:
+1. reads in an env variable, TARGET, from the configuration .yaml file
+2. prints “Hello World: \${TARGET}!“. If TARGET isn’t defined, it will print “NOT SPECIFIED”.
+
+### Procedure
+
+1. Create a new file named `service.yaml` containing the following information.
+
+```yml
+apiVersion: serving.knative.dev/v1 (1)
+kind: Service
+metadata:
+  name: helloworld-go (2)
+  namespace: default (3)
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-samples/helloworld-go (4)
+          env:
+            - name: TARGET (5)
+              value: "Go Sample v1"
+```
+**(1)** Current version of Knative   
+**(2)** The name of the app   
+**(3)** The namespace the app will use   
+**(4)** The URL to the image of the app   
+**(5)** The environment variable printed out by the sample app
+
+**NOTE:** If you’re deploying an image of your own app, update the name of the app and the URL of the image accordingly.
+
+## Creating a private cluster-local Service
+By default services deployed through Knative are published to an external IP
+address, making them public services on a public IP address and with a
+[public URL](./using-a-custom-domain.md).
+
+While this is useful for services that need to be accessible from outside of the
+cluster, frequently you may be building a backend service which should not be
+available off-cluster.
+
+Knative provides two ways to enable private services which are only available
+inside the cluster:
+
+1. To make all services only cluster-local, change the default domain to
+   `svc.cluster.local` by
+   [editing the `config-domain` config map](./using-a-custom-domain.md). This
+   will change all services deployed through Knative to only be published to the
+   cluster, none will be available off-cluster.
+1. To make an individual service cluster-local, the service or route can be
+   labeled in such a way to prevent it from getting published to the external
+   gateway.
+
+### Label a service to be cluster-local
+
+To configure a Service to only be available on the cluster-local network (and
+not on the public Internet), you can apply the
+`serving.knative.dev/visibility=cluster-local` label to the Service or Route
```

Comment:
Since this is "using knative services" you will want to apply it to the Service. The Service will propagate the label to the Route. Any label added to a Route that is owned by a Service will be overwritten.